### PR TITLE
[#2154] fix: Correct the data length and buffer size metrics and logs

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Correct the data length and buffer size metrics and logs

### Why are the changes needed?

Fix: #2154

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs.

### CODE REVIEW

File: server/src/main/java/org/apache/uniffle/server/ShuffleDataFlushEvent.java [hunk 1]
```diff
@@ -37,7 +38,11 @@ public class ShuffleDataFlushEvent {
   private final int shuffleId;
   private final int startPartition;
   private final int endPartition;
+  /** The memory cost size include encoded length */
   private final long size;
+  /** The data size of this shuffle block */
+  private final long length;
```
Comment (jerqi) [inline server/src/main/java/org/apache/uniffle/server/ShuffleDataFlushEvent.java:44]: Could we rename these variables?

Comment (maobaolong) [reply server/src/main/java/org/apache/uniffle/server/ShuffleDataFlushEvent.java]: Yeah, it is a lucky survivor, renamed it.

**Comment by maobaolong** [comment]
@jerqi Thanks for your suggestion, I've addressed it, PTAL.

**Comment by jerqi** [comment]
Could you resolve the conflicts?

**Comment by maobaolong** [comment]
@jerqi Sure, thanks for your remind, I've resolved it, PTAL.

**Review by jerqi — APPROVED** [review]

**Comment by jerqi** [comment]
Could you resolve t

**Comment by maobaolong** [comment]
@jerqi I resolved it, thanks for your remind again! PTAL.

**Review by jerqi — APPROVED** [review]

**Comment by maobaolong** [comment]
@jerqi Thanks for your review!

**Pull request merged**